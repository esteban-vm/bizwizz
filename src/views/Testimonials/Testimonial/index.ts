export { default, type TestimonialProps } from './Testimonial'
