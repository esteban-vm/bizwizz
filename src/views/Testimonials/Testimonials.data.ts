import type { TestimonialProps } from './Testimonial'

const testimonials: TestimonialProps[] = [
  {
    id: 'cdbc1f6c-4153-40f2-a00c-e0957fa2812b',
    name: 'Paul Rodriguez',
    designation: 'Marketer',
    image: { src: '/images/client-1.jpg', height: 536, width: 536 },
    stars: 4,
  },
  {
    id: '74294d20-8111-461a-aa03-afb975deecb8',
    name: 'Ted Johnson',
    designation: 'Life Coach',
    image: { src: '/images/client-2.jpg', height: 523, width: 900 },
    stars: 5,
  },
  {
    id: '827a14ec-c61a-4e3d-b8fc-b1de05c4077b',
    name: 'Paula Lee',
    designation: 'Analyst',
    image: { src: '/images/client-3.jpg', height: 458, width: 592 },
    stars: 3,
  },
]

export default testimonials
