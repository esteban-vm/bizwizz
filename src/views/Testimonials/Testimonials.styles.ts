import clsx from 'clsx'

const styles = {
  wrapper: clsx('m-[0_auto] w-[30%]'),
}

export default styles
