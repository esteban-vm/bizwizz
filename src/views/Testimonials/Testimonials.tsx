import type { FC } from '@/types'
import { Section } from '@/components'
import Carousel from './Carousel'
import Testimonial from './Testimonial'
import testimonials from './Testimonials.data'

const Testimonials: FC = () => {
  return (
    <Section bg='black' id='testimonials' subtitle='People seem to like us!' title='Testimonials'>
      <div className='m-[0_auto] w-3/4 md:w-1/2 lg:w-5/12 xl:w-1/3'>
        <Carousel>
          {testimonials.map((testimonial) => (
            <Testimonial key={testimonial.id} {...testimonial} />
          ))}
        </Carousel>
      </div>
    </Section>
  )
}

export default Testimonials
