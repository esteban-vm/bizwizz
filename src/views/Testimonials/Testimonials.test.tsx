import { render, cleanup, within, screen } from '@/tests'
import Testimonials from './Testimonials'
import testimonials from './Testimonials.data'

describe('Testimonials:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Testimonials />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const title = /testimonials/i
    const subtitle = /people seem to like us!/i
    const testimonialsElem = screen.getByRole('region')
    expect(testimonialsElem).toBeInTheDocument()
    expect(testimonialsElem).toHaveAccessibleName(title)
    expect(testimonialsElem).toHaveAttribute('id', 'testimonials')
    expect(within(testimonialsElem).getByRole('presentation', { hidden: true })).toBeInTheDocument()
    expect(within(testimonialsElem).getByRole('heading', { name: title, level: 2 })).toBeInTheDocument()
    expect(within(testimonialsElem).getByRole('heading', { name: subtitle, level: 3 })).toBeInTheDocument()
    expect(within(testimonialsElem).getByRole('group')).toBeInTheDocument()
    expect(within(testimonialsElem).getAllByRole('article')).toHaveLength(testimonials.length)
  })
})
