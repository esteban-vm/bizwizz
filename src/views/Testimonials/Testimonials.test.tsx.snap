// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testimonials: > should render correctly 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="testimonials_title"
    class="section bg-black"
    id="testimonials"
  >
    <svg
      aria-hidden="true"
      class="wave text-white"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <header>
      <h2
        id="testimonials_title"
      >
        Testimonials
      </h2>
      <h3>
        People seem to like us!
      </h3>
    </header>
    <div
      class="m-[0_auto] w-3/4 md:w-1/2 lg:w-5/12 xl:w-1/3"
    >
      <div
        aria-roledescription="carousel"
        class="flex flex-row items-center justify-between gap-6"
        role="group"
      >
        <svg
          class="cursor-pointer select-none fill-primary hover:opacity-80 active:scale-90"
          role="img"
        >
          <use
            xlink:href="#icon-chevron-left"
          />
        </svg>
        <div
          class="flex flex-col overflow-hidden"
        >
          <input
            class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
            id="slide_1"
            name="slide"
            type="radio"
          />
          <input
            class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
            id="slide_2"
            name="slide"
            type="radio"
          />
          <input
            class="hidden carousel-check:bg-primary carousel-slide-2:translate-x-[calc(-100%/3)] carousel-slide-3:translate-x-[calc(-100%*2/3)]"
            id="slide_3"
            name="slide"
            type="radio"
          />
          <div
            class="order-1 flex justify-center"
          >
            <label
              class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
              for="slide_1"
            />
            <label
              class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
              for="slide_2"
            />
            <label
              class="m-[1.6rem] block h-5 w-5 cursor-pointer rounded-full border-2 border-solid border-primary bg-white"
              for="slide_3"
            />
          </div>
          <div
            class="flex w-[300%] transition-all duration-500"
          >
            <article
              class="m-[3rem_0] flex h-[40rem] w-full flex-col items-start justify-evenly border-[3px] border-solid border-darkgrey-2 bg-darkgrey-2 p-16 text-left hover:border-primary"
              id="cdbc1f6c-4153-40f2-a00c-e0957fa2812b"
            >
              <div
                class="flex flex-row"
              >
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-empty"
                  />
                </svg>
              </div>
              <p
                class="text-3xl text-lightgrey"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Et porro dignissimos quisquam distinctio totam obcaecati laudantium accusamus id ut architecto!
              </p>
              <div
                class="flex items-center gap-10"
              >
                <div
                  class="h-32 w-32 overflow-hidden rounded-full"
                >
                  <img
                    alt="Paul Rodriguez's photo"
                    class="h-full select-none"
                    data-nimg="1"
                    decoding="async"
                    height="536"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fclient-1.jpg&w=1080&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fclient-1.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fclient-1.jpg&w=1080&q=75 2x"
                    style="color: transparent;"
                    width="536"
                  />
                </div>
                <div
                  class="text-2xl"
                >
                  <p
                    class="font-extrabold text-white"
                  >
                    Paul Rodriguez
                  </p>
                  <p
                    class="font-medium text-primary"
                  >
                    Marketer
                  </p>
                </div>
              </div>
            </article>
            <article
              class="m-[3rem_0] flex h-[40rem] w-full flex-col items-start justify-evenly border-[3px] border-solid border-darkgrey-2 bg-darkgrey-2 p-16 text-left hover:border-primary"
              id="74294d20-8111-461a-aa03-afb975deecb8"
            >
              <div
                class="flex flex-row"
              >
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
              </div>
              <p
                class="text-3xl text-lightgrey"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Et porro dignissimos quisquam distinctio totam obcaecati laudantium accusamus id ut architecto!
              </p>
              <div
                class="flex items-center gap-10"
              >
                <div
                  class="h-32 w-32 overflow-hidden rounded-full"
                >
                  <img
                    alt="Ted Johnson's photo"
                    class="h-full select-none"
                    data-nimg="1"
                    decoding="async"
                    height="523"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fclient-2.jpg&w=1920&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fclient-2.jpg&w=1080&q=75 1x, /_next/image?url=%2Fimages%2Fclient-2.jpg&w=1920&q=75 2x"
                    style="color: transparent;"
                    width="900"
                  />
                </div>
                <div
                  class="text-2xl"
                >
                  <p
                    class="font-extrabold text-white"
                  >
                    Ted Johnson
                  </p>
                  <p
                    class="font-medium text-primary"
                  >
                    Life Coach
                  </p>
                </div>
              </div>
            </article>
            <article
              class="m-[3rem_0] flex h-[40rem] w-full flex-col items-start justify-evenly border-[3px] border-solid border-darkgrey-2 bg-darkgrey-2 p-16 text-left hover:border-primary"
              id="827a14ec-c61a-4e3d-b8fc-b1de05c4077b"
            >
              <div
                class="flex flex-row"
              >
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-full"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-empty"
                  />
                </svg>
                <svg
                  class="h-6 w-6 fill-yellow"
                  role="img"
                >
                  <use
                    xlink:href="#icon-star-empty"
                  />
                </svg>
              </div>
              <p
                class="text-3xl text-lightgrey"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Et porro dignissimos quisquam distinctio totam obcaecati laudantium accusamus id ut architecto!
              </p>
              <div
                class="flex items-center gap-10"
              >
                <div
                  class="h-32 w-32 overflow-hidden rounded-full"
                >
                  <img
                    alt="Paula Lee's photo"
                    class="h-full select-none"
                    data-nimg="1"
                    decoding="async"
                    height="458"
                    loading="lazy"
                    src="/_next/image?url=%2Fimages%2Fclient-3.jpg&w=1200&q=75"
                    srcset="/_next/image?url=%2Fimages%2Fclient-3.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fclient-3.jpg&w=1200&q=75 2x"
                    style="color: transparent;"
                    width="592"
                  />
                </div>
                <div
                  class="text-2xl"
                >
                  <p
                    class="font-extrabold text-white"
                  >
                    Paula Lee
                  </p>
                  <p
                    class="font-medium text-primary"
                  >
                    Analyst
                  </p>
                </div>
              </div>
            </article>
          </div>
        </div>
        <svg
          class="cursor-pointer select-none fill-primary hover:opacity-80 active:scale-90"
          role="img"
        >
          <use
            xlink:href="#icon-chevron-right"
          />
        </svg>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
